src/data/astarte_device_sdk_lib.rs

```diff
@@ -97,39 +92,33 @@ impl AstarteDeviceSdkConfigOptions {
         );
 
         if StateRepository::<String>::exists(&registry).await {
-            return registry.read().await.map_err(DeviceSdkError::ReadSecret);
+            return registry
+                .read()
+                .await
+                .map_err(DeviceSdkError::ReadSecret)
+                .map(Credential::secret);
         }
 
         if let Some(token) = &self.pairing_token {
-            return self.register_device(device_id, token, registry).await;
+            // TODO: when will the device be registered?
```

Comment:
Why do you add a TODO here? Is it something to handle in another PR?

Reply:
I forgot it there while rebasing! Just removed